Tutorials/WinForms/Classic/bind-the-data-grid-to-large-data-source.md

```diff
@@ -0,0 +1,294 @@
+* Open the code editor and change the code as follows:
+    <details>
+    <summary>Click to expand</summary>
+
+    ```csharp
+    using DevExpress.Xpo;
+    using DevExpress.XtraBars;
+    using DevExpress.XtraBars.Ribbon;
+    using DevExpress.XtraGrid.Views.Grid;
+    using DxSample.Forms;
+    using System;
+
+    namespace DxSample {
+        public partial class OrdersListForm : RibbonForm {
+            public OrdersListForm() {
+                InitializeComponent();
+            }
+            private void CustomersGridView_RowClick(object sender, RowClickEventArgs e) {
+                if (e.Clicks == 2) {
+                    e.Handled = true;
+                    int orderID = (int)OrdersGridView.GetRowCellValue(e.RowHandle, colOid);
+                    ShowEditForm(orderID);
+                }
+            }
+
+            private void ShowEditForm(int? orderID) {
+                using (EditOrderForm form = new EditOrderForm(orderID)) {
+                    form.ShowDialog(this);
+                    Reload();
+                    OrdersGridView.FocusedRowHandle = OrdersGridView.LocateByValue("Oid", form.OrderID.Value,
+                        rowHandle => OrdersGridView.FocusedRowHandle = (int)rowHandle);
+                }
+            }
+            private void Reload() {
+                OrdersInstantFeedbackView.Refresh();
+            }
+
+            private void btnNew_ItemClick(object sender, ItemClickEventArgs e) {
+                ShowEditForm(null);
+            }
+
+            private void btnDelete_ItemClick(object sender, ItemClickEventArgs e) {
+                using (Session session = new Session()) {
+                    object orderId = OrdersGridView.GetFocusedRowCellValue(colOid);
+                    Order order = session.GetObjectByKey<Order>(orderId);
+                    session.Delete(order);
+                }
+            }
+        }
+    }
+    ```
+    </details>
+
+* Select the **OrdersInstantFeedbackView** component and double click the **ResolveSession** event to add the event handler. Do the same for the **DismissSession** event
+* Put the following code in the event handlers:
+    ```csharp
+    private void OrdersInstantFeedbackView_ResolveSession(object sender, ResolveSessionEventArgs e) {
+        e.Session = new Session()
+    }
+
+    private void OrdersInstantFeedbackView_DismissSession(object sender, ResolveSessionEventArgs e) {
+        e.Session.Session.Dispose()
+    }
+    ```
+## Create the navigation container Form
+* Right-click the project item in the Solution Explorer to invoke the context menu and select the **Add DevExpress Item > New Item** menu item
+* In the DevExpress Template Gallery window select the **WinForms** tab, click the **WinForms Popular UIs > UI-ready Form** item in the navigation control.
+* In the Settings section to the right select the **Tabbed MDI** UI Type, select the **Navigation Container** View Type, and change the **Item Name** to **MainForm**
```

Comment:
Why do you leave the commas out?

Reply:
```suggestion
* In the Settings section to the right, select the **Tabbed MDI** UI Type, select the **Navigation Container** View Type, and change the **Item Name** to **MainForm**
```

Reply:
Why 'to the right'? Are there more than one Setting Sections and you need to distinguish between them?

Reply:
55281262e4ef4ca130b2f36e505c32ea6ddb8d75
929c49e9a98297000913cb11bc755f2dc001bb88